src/vcpkg/base/checks.cpp

```diff
@@ -67,6 +70,11 @@ namespace vcpkg
     [[noreturn]] void Checks::exit_with_code(const LineInfo& line_info, const int exit_code)
     {
         Debug::println(locale_invariant_lineinfo(line_info));
+        // Collect exit code and success telemetry
+        Path file_info{line_info.file_name};
+        get_global_metrics_collector().track_bool(BoolMetric::ExitSuccess, exit_code == EXIT_SUCCESS);
+        get_global_metrics_collector().track_string(
+            StringMetric::ExitCode, fmt::format("{};{};{}", exit_code, file_info.filename(), line_info.line_number));
```

Comment:
We could use multiple columns instead of cramming it all together in a single string (`exit_code`, `exit_file`, `exit_line`).